src/components/popover/Popover.tsx

```diff
@@ -1,151 +1,277 @@
-import * as React from 'react';
 import classNames from 'classnames';
-import { PopperWrapper, PopperWrapperProps } from '@/components/popperWrapper';
-import { BaseProps, filterProps } from '@/utils/types';
-
-type Position = 'top' | 'top-start' | 'top-end' | 'bottom' | 'bottom-start' | 'bottom-end' | 'left' | 'right';
-
-export interface CustomStyle {
-  height?: number | string;
-  minHeight?: number | string;
-  maxHeight?: number | string;
-  width?: number | string;
-  minWidth?: number | string;
-  maxWidth?: number | string;
-}
+import * as React from 'react';
+import { autoUpdate, flip, offset, shift } from '@floating-ui/react-dom';
+import {
+  FloatingFocusManager,
+  FloatingPortal,
+  useClick,
+  useDismiss,
+  useInteractions,
+  useRole,
+  useId,
+  useFloating,
+  useHover,
+  FloatingContext,
+} from '@floating-ui/react-dom-interactions';
+import { motion, AnimatePresence } from 'framer-motion';
 
-const propsList = [
-  'appendToBody',
-  'trigger',
-  'hoverable',
-  'on',
-  'open',
-  'closeOnBackdropClick',
-  'offset',
-  'closeOnScroll',
-] as const;
-type PopperProps = typeof propsList[number];
-
-export interface PopoverProps extends Pick<PopperWrapperProps, PopperProps>, BaseProps {
+export type PositionType =
+  | 'bottom'
+  | 'top'
+  | 'top-start'
+  | 'top-end'
+  | 'bottom-start'
+  | 'bottom-end'
+  | 'left'
+  | 'right';
+
+export type Offset = 'small' | 'medium' | 'large';
+export interface PopoverProps {
   /**
-   * To be rendered in `Popover` component
+   * Element triggering the `Popover`
    */
-  children: React.ReactNode;
```

Comment:
Why are we removing this prop `children` from the prop table but still supporting this inside popover?